apis/v1alpha1/gateway_types.go

```diff
@@ -423,6 +423,39 @@ type RouteBindingSelector struct {
 	Resource string `json:"resource"`
 }
 
+// RouteNamespaces indicate which namespaces Routes should be selected from.
+type RouteNamespaces struct {
+	// NamespaceSelector is a selector of namespaces that Routes should be
+	// selected from. This is a standard Kubernetes LabelSelector, a label query
+	// over a set of resources. The result of matchLabels and matchExpressions
+	// are ANDed. Controllers must not support Routes in namespaces outside this
+	// selector.
+	//
+	// An empty selector (default) indicates that Routes in any namespace can be
+	// selected.
+	//
+	// The OnlySameNamespace field takes precedence over this field. This
+	// selector will only take effect when OnlySameNamespace is false.
+	//
+	// Support: Core
+	//
+	// +optional
+	NamespaceSelector metav1.LabelSelector `json:"namespaceSelector"`
+
+	// OnlySameNamespace is a boolean used to indicate if Route references are
+	// limited to the same Namespace as the Gateway. When true, only Routes
+	// within the same Namespace as the Gateway should be selected.
+	//
+	// This field takes precedence over the NamespaceSelector field. That
+	// selector should only take effect when this field is set to false.
+	//
+	// Support: Core
+	//
+	// +optional
+	// +kubebuilder:default=true
```

Comment:
`true` by default is not the most friendly or good API design. Do we want to consider thinking through this again in a separate issue?

Reply:
That's a good point. I'd prefer to solve this in a separate PR/issue. The only change that's happening to this type in this PR is a move from the gatewayclass_types file to gateway_types file because it's no longer used in GatewayClass.

Reply:
#343